pc/storage/kdtree/kdtree_test.go

```diff
@@ -54,36 +54,73 @@ func TestKDtree(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	return it
+}
+
+func TestKDtree(t *testing.T) {
+	it := createTestPointCloud(t)
 	kdt := New(it)
 
-	expectedTree := &KDTree{
-		Vec3RandomAccessor: it,
-		root: &node{
-			children: [2]*node{
-				&node{
-					children: [2]*node{
-						&node{id: 5, dim: 2},
-						&node{id: 1, dim: 2},
+	t.Run("Equal", func(t *testing.T) {
+		expectedTree := &KDTree{
+			Vec3RandomAccessor: it,
+			root: &node{
+				children: [2]*node{
+					&node{
+						children: [2]*node{
+							&node{id: 5, dim: 2},
+							&node{id: 1, dim: 2},
+						},
+						id:  4,
+						dim: 1,
+					},
+					&node{
+						children: [2]*node{
+							&node{id: 2, dim: 2},
+							&node{id: 6, dim: 2},
+						},
+						id:  0,
+						dim: 1,
 					},
-					id:  4,
-					dim: 1,
 				},
-				&node{
-					children: [2]*node{
-						&node{id: 2, dim: 2},
-						&node{id: 6, dim: 2},
+				id:  3,
+				dim: 0,
+			},
+		}
+		if !kdt.Equal(expectedTree) {
+			t.Fatalf("%v and %v must be equal", expectedTree, kdt)
+		}
```

Comment:
To use `Kdtree.Equal()` in tests, `Kdtree.Equal()` must be tested separately.
(If `Kdtree.Equal()` were not working correctly, all other tests relying on it may be wrong)